Doc/library/socket.rst

```diff
@@ -631,6 +647,18 @@ Constants
 
    .. versionadded:: 3.12
 
+.. _socket-ethernet-types:
+
+.. data:: ETHERTYPE_ARP
+          ETHERTYPE_IP
+          ETHERTYPE_IPV6
+          ETHERTYPE_VLAN
+
+   IEEE 802.3 protocol numbers.
```

Comment:
Would be nice to be a little more specific here, and include a reference (e.g. a manpage or online documentation) where they are documented further.

Reply:
Also, is there a reason this isn't included right before or after ``ETH_P_ALL``, given their relation?

Reply:
`ETHERTYPE_*` are just the Ethernet type codes for various packet types - it can be used for packet socket creation but also for other things such as manually crafting a network packet.
`ETH_P_ALL` is Linux specific for packet socket creation to capture all frame, it has no meaning other than that.

`ETH_P_ALL` is documented right after `AF_PACKET`, because they go together.
`ETHERTYPE_*` are unrelated to that so I included them in a different place.